lib/omise/OmiseSearch.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+require_once dirname(__FILE__).'/res/OmiseApiResource.php';
+
+/**
+ * This class is not intended to be used directly from client code.
+ *
+ * @see OmiseCharge::search()
+ */
+class OmiseSearch extends OmiseApiResource
+{
+    const ENDPOINT = 'search';
+
+    private $dirty = true;
+    private $scope = null;
+    private $query = '';
+    private $conditions = array();
+    private $page = 1;
+    private $order = 'chronological';
+
+    public static function scope($scope)
+    {
+        return new OmiseSearch($scope);
+    }
+
+    protected function __construct($scope, $publickey = null, $secretkey = null)
+    {
+        parent::__construct($publickey, $secretkey);
+        $this->dirty = true;
+        $this->scope = $scope;
+    }
+
+    public function query($query)
+    {
+        $this->dirty = true;
+        $this->query = $query;
+        return $this;
+    }
+
+    public function where(array $conditions = array())
+    {
+        $this->dirty = true;
+        $this->conditions = $conditions;
+        return $this;
+    }
+
+    public function page($page)
+    {
+        $this->dirty = true;
+        $this->page = $page;
+        return $this;
+    }
+
+    public function order($order)
+    {
+        if (!in_array($order, array('chronological', 'reverse_chronological'))) {
```

Comment:
@guzzilar I also will remove check on `order` so that the library can support new `order` modes without the need to change library side.